lilypad/_utils/middleware.py

```diff
@@ -44,17 +44,14 @@ def custom_context_manager(
                 "lilypad.project_uuid": str(new_project_uuid)
                 if new_project_uuid
                 else "",
-                "lilypad.function_name": fn.__name__,
-                "lilypad.version_num": version.version_num
-                if version.version_num
-                else -1,
-                "lilypad.version_uuid": str(version.uuid),
-                "lilypad.arg_types": json.dumps(arg_types),
-                "lilypad.arg_values": json.dumps(arg_values),
-                "lilypad.lexical_closure": version.function.code,
-                "lilypad.prompt_template": version.prompt.template
-                if version.prompt
-                else prompt_template or "",
+                "lilypad.type": "generation",
+                "lilypad.generation.uuid": str(version.uuid),
+                "lilypad.generation.name": fn.__name__,
+                "lilypad.generation.signature": version.signature,
+                "lilypad.generation.code": version.code,
+                "lilypad.generation.arg_types": json.dumps(arg_types),
+                "lilypad.generation.arg_values": json.dumps(arg_values),
+                "lilypad.generation.prompt_template": prompt_template or "",
```

Comment:
why does generation have prompt template, shouldnt this be lilypad.prompt.template?

Reply:
this is for mirascope calls that will have prompt templates at the generation level for our first class support

Reply:
got it